src/DataExplorer/DataExplorerVisualizations.jsx

```diff
@@ -117,6 +175,9 @@ class DataExplorerVisualizations extends React.Component {
     if (buttonConfig.type === 'manifest' && this.state.nodeIds.length > 0) {
       buttonTitle = `${buttonConfig.title} (${humanizeNumber(this.state.manifestEntryCount)})`;
     }
+    if (buttonConfig.type === 'export-to-workspace' && this.state.exportInProgress) {
```

Comment:
there is a new property on the button called `isPending` that I think accomplishes all this functionality for you - if you set `<Button ... isPending={this.state.exportInProgress}` it will disable the button and add a spinner to the button to indicate something is happening